client/src/components/HabitItem.jsx

```diff
@@ -1,53 +1,74 @@
-import React, { useState } from 'react';
-import axiosInstance from '../api/axiosInstance.js';
+import React, { useState, useEffect } from 'react';
+import axios from '../api/axiosInstance';
 import './Habit.css';
 
-function HabitItem({ habit, onEdit, onDelete, onToggle, isCompleted }) {
-    const [streak, setStreak] = useState(habit.streak);
-
-    const handleToggle = () => {
-    const newStatus = !isCompleted;
-
-    // Optimistically update streak
-    setStreak((prev) => newStatus ? prev + 1 : Math.max(prev - 1, 0));
-
-    axiosInstance
-    .post('/api/habit-logs', {
-    habitId: habit.id,
-    completed: newStatus,
-    })
-    .then(() => {
-    if (onToggle) onToggle(habit.id, isCompleted);
-    })
-    .catch((error) => {
-    console.error('Error logging habit completion:', error);
-    setStreak((prev) => newStatus ? prev - 1 : prev + 1); // revert streak
-    });
-    };
-
-    return (
-    <div className="habit-card">
-    <div className="habit-header">
-    <h3>{habit.title}</h3>
-    <label className="toggle-switch">
-    <input
-    type="checkbox"
-    checked={isCompleted}
-    onChange={handleToggle}
-    />
-    <span className="slider"></span>
-    </label>
-    </div>
-    {habit.description && <p>{habit.description}</p>}
-    <p className="habit-meta">
-    Streak: {streak} | {habit.isActive ? 'Active' : 'Inactive'}
-    </p>
-    <div className="habit-actions">
-    <button onClick={() => onEdit(habit)}>Edit</button>
-    <button onClick={() => onDelete(habit.id)}>Delete</button>
-    </div>
-    </div>
-    );
-    }
+function HabitItem({ habit, isCompleted, onEdit, onDelete, onToggle }) {
+const [streak, setStreak] = useState(habit.streak || 0);
+const [lastCompletedDate, setLastCompletedDate] = useState(habit.lastCompletedDate || null);
+const [toggleDisabled, setToggleDisabled] = useState(false);
+
+useEffect(() => {
+// Check if last completed date is today
```

Comment:
Can you format this file with appropriate indentation (like the other jsx files)? It is a bit hard to read without indentation.